python_modules/libraries/dagster-dbt/dagster_dbt/asset_utils.py

```diff
@@ -445,6 +445,23 @@ def group_from_dbt_resource_props_fallback_to_directory(
     return fqn[1]
 
 
+def default_owners_from_dbt_resource_props(
+    dbt_resource_props: Mapping[str, Any],
+) -> Optional[Sequence[str]]:
+    dagster_metadata = dbt_resource_props.get("meta", {}).get("dagster", {})
+    owners_config = dagster_metadata.get("owners")
```

Comment:
The values under this "owners" config might not match our expected owners format, right? So this could cause some dbt projects to start erroring?

Reply:
This is user-specified Dagster metadata on a dbt model. I doubt that they were specifying an `owners` meta field under that before this change.

If the concern is that they could instead provide something like:

```yaml
meta:
  dagster:
    owner: ["not_in_correct_owner_format"]
```    
   
Yeah that's a possibility. But I was assuming that this would be caught at definition time, so it's not a problem.

Errors would only arise if the user opted-in and put an incorrect value.

Reply:
Ahh got it. I wonder how much of a convention using "owner" config is? It's their first example on their meta docs page: https://docs.getdbt.com/reference/resource-configs/meta#examples.

Ideally, users would not need to specify this twice.

Reply:
I think their documentation is out of date. Although they have this as an example with dbt meta, they have a "groups" capability that supports this owner field natively.

By default, we retrieve this dbt group information (released in `dbt-core==1.5`). Not the meta field. The dbt group is the source of truth for groups/owners.

Reply:
> The dbt group is the source of truth for groups/owners.

Just because it's possible to set an owner on a group and set a group on a model doesn't necessarily indicate to me that this is only canonical dbt way to set an owner on the model.

From what I can tell, other data catalogs seem to scrape from the owner meta field:
- https://docs.open-metadata.org/v1.3.x/connectors/ingestion/workflows/dbt/ingest-dbt-owner
- https://datahubproject.io/docs/generated/ingestion/sources/dbt/ (ctrl-f for "owner_extraction_pattern")

Reply:
It feels weird to reify that convention into supported configuration, considering that the convention is not standardized.

Like you said, if the owner there is not an email, what should happen? Do we error? Do we silently ignore it and allow it to continue?

For now, I would instead take the safe configuration, where we know an email is available. If we get feedback from users that this top level owners field is something they want supported, then we can figure out those semantics.

Reply:
> It feels weird to reify that convention into supported configuration, considering that the convention is not standardized.

FWIW that's what those linked data catalog tools seem to do.

> Like you said, if the owner there is not an email, what should happen? Do we error? Do we silently ignore it and allow it to continue?

I think this points to a flaw with our owner system that we probably at least eventually need to patch. We've been relatively conservative here because it's the easiest thing to do, but I think we're ultimately going to support more flexible values to accommodate systems we wrap that have more flexible values.

> If we get feedback from users that this top level owners field is something they want supported, then we can figure out those semantics.

Imagining a future world where we've done that:
- Then there would be two different ways to set owners, one under "dagster" and one not.
- I suppose the way we'd message it is that the top-level owners field is the default one to use, but the "dagster" one is the one you use if you want your owners in Dagster to be different than your dbt owners?

Ok so having thought this through, that sentence above convinces me that we could go ahead with the approach in this PR for now and still have room to add support for a top-level owners field at a later date. Thoughts?

Reply:
> FWIW that's what those linked data catalog tools seem to do.

My qualms here are that these linked data catalog tools just ingest owner as any arbitrary string. 

In Dagster, we don't do that. Like you said, we're conservative here:
- We either want the string as an email
- Or, we either want the string to be prefixed with `team:`.

If we want to get around this for dbt, then if they pass in an arbitrary string to their top-level owners field in meta, then we should interpret it as a "team" on our side. But I think this would be a confusing assumption.

So that we don't have to make an assumption about how the owner is represented, I left it out.

> Imagining a future world where we've done that:
>
> Then there would be two different ways to set owners, one under "dagster" and one not.
I suppose the way we'd message it is that the top-level owners field is the default one to use, but the "dagster" one is the one you use if you want your owners in Dagster to be different than your dbt owners?

Yeah, I agree with this. The ideal default hierarchy should be owners, as specified in:
1. The `+meta.dagster.owners` field, which supersedes
2. The `+meta.owner` field, which supersedes
3. The `+group` field, which refers to a `+group.email`.

And for users who want to customize how this done (e.g. maybe their owners already live in something like `+meta.asset_lord` or something, or maybe they want to parse their owner in a certain way), they always have the option to override `DagsterDbtTranslator.get_owners`.